<launch>
    <arg name="camera" default="et7"/>

    <node pkg="aruco_mapping_ecsam" type="aruco_mapping" name="aruco_mapping_et7" output="screen">
        <remap from="image_raw" to="$(arg camera)/scene_camera/image_raw"/>
        <remap from="aruco_poses" to="$(arg camera)/aruco_poses"/>
        <param name="camera_frame" type="string" value="$(arg camera)_aruco_scene_camera_frame"/>
        <param name="marker_origin_frame" type="string" value="$(arg camera)_aruco_marker_origin_frame"/>
        <param name="calibration_file" type="string"
               value="$(find aruco_mapping_ecsam)/config/$(arg camera)_scene_camera.ini"/>
        <param name="num_of_markers" type="int" value="10"/>
        <param name="marker_size" type="double" value="0.124"/>
        <param name="space_type" type="string" value=""/>
        <!--    <param name="space_type" type="string" value="plane" />-->
    </node>

    <node pkg="aruco_mapping_ecsam" type="root_marker_static_tf_broadcaster"
          name="root_marker_static_tf_broadcaster_$(arg camera)" output="screen">
        <remap from="aruco_poses" to="$(arg camera)/aruco_poses"/>
        <param name="root_frame" type="string" value="root"/>
        <param name="marker_origin_frame" type="string" value="$(arg camera)_aruco_marker_origin_frame"/>
    </node>


    <node pkg="tf2_ros" type="static_transform_publisher" name="et7_arcuro_frame_transformed"
          args="0 0 0 0 -0.02 -0.105 et7_aruco_scene_camera_frame et7_scene_camera_frame"/>

</launch>
